docs/custom/error-handlers.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: error_handlers
+title: Custom error handlers
+---
+
+:::caution
+This AMF functionality is currently a work in progress. The following features are being developed:
+- Support for custom error handling in parsing, validation and emission stages
+- Javascript support
+:::
+
+In this documentation you'll learn what is an error handler in AMF and how to create one yourself with code examples.
+
+
+# What's an AMF error handler
+
+An error handler is a mechanism that lets you handle any errors that arise while using AMF.
+AMF allows you to define custom error handlers that you can customize and plug into AMF's resolution stage (and in the future, all stages).
+
+There are 3 types of errors in AMF, denoted by the `amf.core.validation.SeverityLevels` object:
+
+- **Info**
+  - This severity level points to something that may or may not be wrong
+  - Usually depends on user intent or use case
+- **Warning**
+  - This severity level indicates to the user that something is not correct, and should be changed
+  - Usually warns about future changes, using deprecated tools, or errors that will become violations the future
+- **Violation**
+  - This severity level indicates that something is definitely wrong
+  - Indication that something is not compliant with its specification
+  - This severity makes the AMF graph model invalid
+
+# Default error handling process
```

Comment:
## How error handling works in AMF

AMF transforms each error into an `ValidationResult`, and collects all generated results in a `ValidationReport`. After the [validation stage](../AMF/using-amf/amf_validation#validation-examples), AMF returns the `ValidationReport`.

```scala title="amf.core.validation.core.ValidationReport"
trait ValidationResult {
  def message: Option[String]
  def path: String
  def sourceConstraintComponent: String
  def focusNode: String
  def severity: String
  def sourceShape: String
}
trait ValidationReport {
  def conforms: Boolean
  def results: List[ValidationResult]
}
```